src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Color DEFAULT_COLOR = Color.WHITE;
+    private static final int DEFAULT_RADIUS = 10;
+    private static final int MAX_RANGE = 20;
+    private final Random random = new Random();
+    private final Circle circle = new Circle();
+    private final Square square = new Square();
+    private final RightTriangle rightTriangle = new RightTriangle();
+    private final Rectangle rectangle = new Rectangle();
+    private final IsoscelesTrapezoid isoscelesTrapezoid = new IsoscelesTrapezoid();
+    private final ColorSupplier colorSupplier = new ColorSupplier(random);
+    private final Figure[] figures = {circle, square, rightTriangle, rectangle, isoscelesTrapezoid};
+
+    public Figure getRandomFigure() {
+        int randomNameFigure = random.nextInt(figures.length);
```

Comment:
Here is the only place where you need your `figures`. Instead, you can do next:
```suggestion
        int randomNameFigure = random.nextInt(FIGURES_COUNT);
```

Declate `FIGURES_COUNT` as a constant an assign there a corresponding value.